From 0a8de08994494517d95139dbd174ab47e523db21 Mon Sep 17 00:00:00 2001
From: Alejandro Abdelnur <tucu@apache.org>
Date: Thu, 3 Jan 2013 11:49:55 +0000
Subject: [PATCH 1240/1357] MR1: MAPREDUCE-2217. The expire launching task should cover the UNASSIGNED task. (schen and kkambatl via tucu)

git-svn-id: https://svn.apache.org/repos/asf/hadoop/common/branches/branch-1@1428304 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit 6a4e45250b89b01920e6d8ab86623c08304c2229)

Reason: Tasks hang in the UNASSIGNED state on faulty TTs
Ref: CDH-8889
Author: Karthik Kambatla and Scott Chen
---
 .../org/apache/hadoop/mapred/JobTracker.java       |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/mapred/org/apache/hadoop/mapred/JobTracker.java b/src/mapred/org/apache/hadoop/mapred/JobTracker.java
index ad5f9be..5559fda 100644
--- a/src/mapred/org/apache/hadoop/mapred/JobTracker.java
+++ b/src/mapred/org/apache/hadoop/mapred/JobTracker.java
@@ -4254,8 +4254,10 @@ public class JobTracker implements MRConstants, JTProtocols, JobTrackerMXBean {
       report.setTaskTracker(trackerName);
       TaskAttemptID taskId = report.getTaskID();
       
-      // expire it
-      expireLaunchingTasks.removeTask(taskId);
+      // don't expire the task if it is not unassigned
+      if (report.getRunState() != TaskStatus.State.UNASSIGNED) {
+        expireLaunchingTasks.removeTask(taskId);
+      }
       
       JobInProgress job = getJob(taskId.getJobID());
       if (job == null) {
-- 
1.7.0.4

